File: system-test/sec-tests/typescript/sec_24_mass_assignment.ts
// SEC-24: Mass Assignment
// Test: TC-CR-030
import express, { Request, Response } from 'express';

const app = express();

interface UserInput {
  username: string;
  email: string;
  isAdmin?: boolean;
}

app.post('/users', async (req: Request, res: Response) => {
  const user = await User.create(req.body);  // Attacker can set isAdmin
  res.json(user);
});

File: system-test/sec-tests/python/sec_24_mass_assignment.py
# SEC-24: Mass Assignment
# Test: TC-CR-030
from flask import Flask, request
from models import User, db

app = Flask(__name__)

@app.route('/users', methods=['POST'])
def create_user():
    data = request.get_json()
    # Attacker can set is_admin=true
    user = User(**data)
    db.session.add(user)
    db.session.commit()
    return 'created'
